supabase/migrations/20240501085903_handle_new_user_update.sql

```diff
@@ -0,0 +1,30 @@
+-- extends function added in 20240331171034_public_users.sql
+CREATE OR REPLACE FUNCTION public.handle_new_user()
+    RETURNS trigger
+    LANGUAGE plpgsql
+    SECURITY DEFINER
+    SET search_path TO 'public'
+AS $function$
+BEGIN
+    INSERT INTO public.users (id, email, name, avatar_url, created_at, updated_at)
+    VALUES (
+               new.id,
+               new.email,
+               new.raw_user_meta_data ->> 'name',
+               new.raw_user_meta_data ->> 'picture',
+               new.created_at,
+               new.updated_at
+           );
+
+    -- If user has accepted an invitation(s) before signing up, add them to the project(s)
```

Comment:
Is it possible to add this block below as a separate function?

So that the migration file basically contains only changes.

Reply:
I could create two standalone functions, then wrap them in another function that would be triggered.

```
CREATE OR REPLACE FUNCTION public.xxx()
RETURNS TRIGGER AS $$
BEGIN
    PERFORM public.handle_new_user();

    PERFORM public.create_project_member();
END;
$$ LANGUAGE plpgsql;
```

```
CREATE TRIGGER on_auth_user_created
AFTER INSERT ON auth.users
FOR EACH ROW EXECUTE PROCEDURE public.xxx()
```

But I will stick with my current solution.